src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -271,27 +283,71 @@ ErrorIfModifyQueryNotSupported(Query *queryTree)
 				continue;
 			}
 
-			if (contain_mutable_functions((Node *) targetEntry->expr))
+			if (commandType == CMD_UPDATE &&
+				contain_volatile_functions((Node *) targetEntry->expr))
 			{
-				hasNonConstTargetEntryExprs = true;
+				ereport(ERROR, (errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+								errmsg("functions used in UPDATE queries on distributed "
```

Comment:
Yeah, the problem is that if we say `UPDATE tbl SET value = random() WHERE key = 10` a non-distributed postgres will give every effected row a different value, whereas evaluating on the master would give every row the same value.